airflow/api/common/experimental/mark_tasks.py

```diff
@@ -312,8 +374,6 @@ def set_dag_run_state_to_success(dag, execution_date, commit=False, session=None
         _set_dag_run_state(dag.dag_id, execution_date, State.SUCCESS, session)
 
     # Mark all task instances of the dag run to success.
-    for task in dag.tasks:
-        task.dag = dag
```

Comment:
Could we remove this code?

Reply:
Not sure. @mik-laj -> maybe you can take a look, you know much better now how changing state works and whether this change will be ok.